src/Domain/ElementMap.hpp

```diff
@@ -0,0 +1,134 @@
+    }
+    return source_point;
+  }
+
+  template <typename T>
+  Tensor<T, tmpl::integral_list<std::int32_t, 2, 1>,
+         index_list<SpatialIndex<Dim, UpLo::Up, Frame::Logical>,
+                    SpatialIndex<Dim, UpLo::Lo, TargetFrame>>>
+  inv_jacobian(tnsr::I<T, Dim, Frame::Logical> source_point) const noexcept {
+    for (size_t d = 0; d < Dim; ++d) {
+      source_point.get(d) = source_point.get(d) * gsl::at(map_slope_, d) +
+                            gsl::at(map_offset_, d);
+    }
+    auto inv_jac = block_map_->inv_jacobian(source_point);
+    for (size_t d = 0; d < Dim; ++d) {
+      for (size_t i = 0; i < Dim; ++i) {
+        inv_jac.get(d, i) *= gsl::at(inverse_jacobian_, d);
+      }
+    }
+    return inv_jac;
+  }
+
+  template <typename T>
+  Tensor<T, tmpl::integral_list<std::int32_t, 2, 1>,
+         index_list<SpatialIndex<Dim, UpLo::Up, TargetFrame>,
+                    SpatialIndex<Dim, UpLo::Lo, Frame::Logical>>>
+  jacobian(tnsr::I<T, Dim, Frame::Logical> source_point) const noexcept {
+    for (size_t d = 0; d < Dim; ++d) {
+      source_point.get(d) = source_point.get(d) * gsl::at(map_slope_, d) +
+                            gsl::at(map_offset_, d);
+    }
+    auto jac = block_map_->jacobian(source_point);
+    for (size_t d = 0; d < Dim; ++d) {
+      for (size_t i = 0; i < Dim; ++i) {
+        jac.get(i, d) *= gsl::at(jacobian_, d);
+      }
+    }
+    return jac;
+  }
+
+  // clang-tidy: do not use references
+  void pup(PUP::er& p) noexcept;  // NOLINT
+
+ private:
+  std::unique_ptr<CoordinateMapBase<Frame::Logical, TargetFrame, Dim>>
+      block_map_{nullptr};
+  ElementId<Dim> element_id_{};
+  // length_of_range_[i] = segment_ids[i].endpoint(Side::Upper) -
+  //                       segment_ids[i].endpoint(Side::Lower)
+  std::array<double, Dim> length_of_range_{
+      make_array<Dim>(std::numeric_limits<double>::signaling_NaN())};
+  // length_of_range_[i] = segment_ids[i].endpoint(Side::Upper) +
+  //                       segment_ids[i].endpoint(Side::Lower)
+  std::array<double, Dim> sum_of_endpoints_{
+      make_array<Dim>(std::numeric_limits<double>::signaling_NaN())};
+  // length_of_range_ / length_of_domain_ = 0.5 * length_of_range_
+  std::array<double, Dim> map_slope_{
+      make_array<Dim>(std::numeric_limits<double>::signaling_NaN())};
+  // (a_ + b_) / (B_ - A_) = (a_ + b_) / length_of_domain_ = 0.5 * (a_ + b_)
+  std::array<double, Dim> map_offset_{
+      make_array<Dim>(std::numeric_limits<double>::signaling_NaN())};
+  // length_of_domain_ / length_of_range_ = Dim.0 / length_of_range_
+  std::array<double, Dim> map_inverse_slope_{
+      make_array<Dim>(std::numeric_limits<double>::signaling_NaN())};
+  // - (b_ + a_) / length_of_range_
+  std::array<double, Dim> map_inverse_offset_{
+      make_array<Dim>(std::numeric_limits<double>::signaling_NaN())};
+  // length_of_range_ / length_of_domain_ = 0.5 * length_of_range_
+  std::array<double, Dim> jacobian_{
```

Comment:
A brief comment that the Jacobian is diagonal and so these are the diagonal entries would make things clearer.